src/EFCore.SqlServer.NTS/Query/Internal/SqlServerNetTopologySuiteDbFunctionsMethodCallTranslator.cs

```diff
@@ -0,0 +1,63 @@
+///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+///     any release. You should only use it directly in your code with extreme caution and knowing that
+///     doing so can result in application failures when updating to a new Entity Framework Core release.
+/// </summary>
+public class SqlServerNetTopologySuiteDbFunctionsMethodCallTranslator : IMethodCallTranslator
+{
+    private static readonly MethodInfo CurveToLineMethodInfo = typeof(SqlServerNetTopologySuiteDbFunctionsExtensions)
+        .GetMethod(nameof(SqlServerNetTopologySuiteDbFunctionsExtensions.CurveToLine), new[] { typeof(DbFunctions), typeof(Geometry) })!;
+
+    private readonly IRelationalTypeMappingSource _typeMappingSource;
+    private readonly ISqlExpressionFactory _sqlExpressionFactory;
+
+    /// <summary>
+    ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+    ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+    ///     any release. You should only use it directly in your code with extreme caution and knowing that
+    ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+    /// </summary>
+    public SqlServerNetTopologySuiteDbFunctionsMethodCallTranslator(
+        IRelationalTypeMappingSource typeMappingSource,
+        ISqlExpressionFactory sqlExpressionFactory)
+    {
+        _typeMappingSource = typeMappingSource;
+        _sqlExpressionFactory = sqlExpressionFactory;
+    }
+
+    /// <summary>
+    ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+    ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+    ///     any release. You should only use it directly in your code with extreme caution and knowing that
+    ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+    /// </summary>
+    public virtual SqlExpression? Translate(SqlExpression? instance, MethodInfo method, IReadOnlyList<SqlExpression> arguments, IDiagnosticsLogger<DbLoggerCategory.Query> logger)
+    {
+        if (method.Equals(CurveToLineMethodInfo))
+        {
+            return _sqlExpressionFactory.Function(
+                arguments[1],
+                "STCurveToLine",
+                Enumerable.Empty<SqlExpression>(),
+                nullable: true,
+                instancePropagatesNullability: true,
+                argumentsPropagateNullability: Enumerable.Empty<bool>(),
+                method.ReturnType,
+                _typeMappingSource.FindMapping(
+                    method.ReturnType,
+                    ExpressionExtensions.InferTypeMapping(arguments[1])!.StoreType));
```

Comment:
Any reason to use InferTypeMapping here? I think it's useful only when there are multiple things to infer from...

```suggestion
                    arguments[1]!.StoreType));
```

Reply:
Thanks, I thought that seemed a bit overcomplicated. Any reason I can't just use argument[1].TypeMapping here instead of FindMapping?